inferno-core/src/Inferno/Infer.hs

```diff
@@ -1031,17 +1034,50 @@ infer expr =
                             LText _ -> typeText
                         )
                     POne _ p -> do
-                      (t, vars) <- mkPatConstraint p
+                      (t, vars, cs) <- mkPatConstraint p
                       meta <- lookupEnv patLoc $ Left oneHash
                       attachTypeToPosition patLoc meta {ty = (Set.empty, ImplType Map.empty $ t .-> TOptional t)}
-                      return (TOptional t, vars)
+                      return (TOptional t, vars, cs)
                     PEmpty _ -> do
                       meta <- lookupEnv patLoc $ Left emptyHash
                       let (_, ImplType _ t) = ty meta
                       attachTypeToPosition patLoc meta
-                      return (t, [])
+                      return (t, [], Set.empty)
+                    PArray _ [] _ -> do
+                      tv <- fresh
+                      let t = TArray tv
+                      attachTypeToPosition
+                        patLoc
+                        TypeMetadata
+                          { identExpr = patternToExpr $ bimap (const ()) (const ()) pattern,
```

Comment:
One thing that might matter eventually, if `-XPatternSynonyms` is on (which it is in some `inferno` modules), this is going to break because `pattern` becomes a reserved keyword